client/lib/plugins/actions.js

```diff
@@ -64,11 +44,75 @@ function queueSitePluginAction( action, siteId, pluginId, callback ) {
 		} );
 	} else {
 		_actionsQueueBySite[ siteId ] = [];
-		action( siteId, pluginId, next.bind( this, callback ) );
+		action( next.bind( this, callback ) );
 	}
-}
+};
+
+const queueSitePluginActionAsPromise = ( action, siteId, pluginId, callback ) => {
+	var promise = new Promise( ( resolve, reject ) => {
+		queueSitePluginAction( action, siteId, pluginId, ( error, data ) => {
+			if ( callback ) {
+				callback( error, data );
+			}
+			if ( error ) {
+				reject( error );
+			}
+			resolve( data );
+		} );
+	} );
+
+	return promise;
+};
+
+const getSolvedPromise = ( dataToPass ) => {
+	return new Promise( resolve => resolve( dataToPass ) );
+};
+
+const getRejectedPromise = ( errorToPass ) => {
+	return new Promise( ( resolve, reject ) => reject( errorToPass ) );
+};
+
+/**
+ * Return plugin id depending if the site is a jetpack site
+ *
+ * @param {Object} site - site object
+ * @param {Object} plugin - plugin object
+ * @return {String} plugin if
+ */
+const getPluginId = ( site, plugin ) => {
```

Comment:
I'm not 100% sure because I'm not familiar with the work being done on the new .com plugins, but I think we could remove this and just use plugin.id, since (I believe) this methods are going to be used just with jetpack sites